core/src/main/java/com/graphhopper/routing/util/AbstractFlagEncoder.java

```diff
@@ -238,102 +252,70 @@ public long handleNodeTags(ReaderNode node) {
                     return 0;
 
                 if (node.hasTag(res, restrictedValues))
-                    return directionBitMask;
+                    return encoderBitMask;
             }
 
             if (blockByDefault)
-                return directionBitMask;
+                return encoderBitMask;
         }
 
         // In case explicit flag ford=no, don't block
         if (blockFords
                 && (node.hasTag("highway", "ford") || node.hasTag("ford"))
                 && !node.hasTag(restrictions, intendedValues)
                 && !node.hasTag("ford", "no")) {
-            return directionBitMask;
-
+            return encoderBitMask;
         }
 
         return 0;
     }
 
     @Override
-    public InstructionAnnotation getAnnotation(long flags, Translation tr) {
+    public InstructionAnnotation getAnnotation(IntsRef ints, Translation tr) {
         return InstructionAnnotation.EMPTY;
     }
 
-    /**
-     * Swapping directions means swapping bits which are dependent on the direction of an edge like
-     * the access bits. But also direction dependent speed values should be swapped too. Keep in
-     * mind that this method is performance critical!
-     */
-    public long reverseFlags(long flags) {
-        long dir = flags & directionBitMask;
-        if (dir == directionBitMask || dir == 0)
-            return flags;
-
-        return flags ^ directionBitMask;
-    }
-
-    /**
-     * Sets default flags with specified access.
-     */
-    public long flagsDefault(boolean forward, boolean backward) {
-        long flags = speedEncoder.setDefaultValue(0);
-        return setAccess(flags, forward, backward);
-    }
-
     @Override
-    public long setAccess(long flags, boolean forward, boolean backward) {
-        return setBool(setBool(flags, K_BACKWARD, backward), K_FORWARD, forward);
-    }
-
-    @Override
-    public long setSpeed(long flags, double speed) {
+    public IntsRef setSpeed(IntsRef ints, double speed) {
         if (speed < 0 || Double.isNaN(speed))
             throw new IllegalArgumentException("Speed cannot be negative or NaN: " + speed
-                    + ", flags:" + BitUtil.LITTLE.toBitString(flags));
+                    + ", flags:" + BitUtil.LITTLE.toBitString(ints));
 
-        if (speed < speedEncoder.factor / 2)
-            return setLowSpeed(flags, speed, false);
+        if (speed < speedFactor / 2) {
+            averageSpeedEnc.setDecimal(false, ints, 0);
+            accessEnc.setBool(false, ints, false);
+            // accessEnc.setBool(true, ints, false);
```

Comment:
Why was this commented out? I just removed this comment and this fixed `CarFlagEncoderTest#testSetSpeed0_issue367`. Also it seems reasonable that if setting speed ~= 0 that access is false, except for Encoders with speeds per direction.

This was commented out [here](https://github.com/graphhopper/graphhopper/commit/8dd658c5ae48e8a690a90ed6c9135d4ae7d7479a). The number of failing tests did not change (except for `CarFlagEncoderTest#testSetSpeed0_issue367`).

Reply:
speed 0 vs. accessibility is a problem, because they usually depend on each other but not always (traffic jam). If we would have one speed value per direction we wouldn't have this problem but this would reduce available bits. Though problem.

Reply:
> Lets assume we have one speed value per direction (2 per edge) then the previous state was correct

Are you sure that the previous state was correct? The previous state always set the forward access to false and the backward access got commented out, but was in there before. Shouldn't we first try not to change any concepts (ideally leaving most tests as they are) and just get it working before we start changing concepts?

Reply:
Yes, avoiding changing concepts for the deprecated FlagEncoder API makes 100% sense.